jetty-jaspi/src/main/java/org/eclipse/jetty/security/jaspi/modules/FormAuthModule.java

```diff
@@ -173,6 +173,7 @@ public AuthStatus validateRequest(MessageInfo messageInfo, Subject clientSubject
                     }
 
                     response.setContentLength(0);
+                    Request.getBaseRequest(request).getHttpChannel().ensureContentConsumedOrConnectionClose();
```

Comment:
I would put this code _inside_ `sendRedirect()` to: A) avoid duplicating it all over the places and B) avoid `Request.getBaseRequest(request)` since I'm already inside the Jetty `Response` where I can easily access the `HttpChannel`.

Reply:
It can't go inside of `sendRedirect()` as it is perfectly OK for an application to send a redirect and then consume the content.
We can only do it on our own sendRedirect calls.  We could have our own method for `consumeAllAndSendRedirect`?